### PR TITLE
Adding Support for Non-Parametric Layernorm

### PR DESCRIPTION
**Pull request by aflah02**
Hi
Based on our discord discussion I've added non-parametric layernorm as described in the OLMo paper.

### CODE REVIEW

File: megatron/model/norms.py [hunk 1]
```diff
@@ -33,6 +33,12 @@ def get_norm(neox_args):
             norm = MixedFusedLayerNorm
         else:
             norm = LayerNorm
+    elif neox_args.norm == "non_parametric_layernorm":
+        eps = neox_args.layernorm_epsilon
+        if neox_args.layernorm_fusion:
+            raise ValueError(f"neox_args.layernorm_fusion not supported for non_parametric_layernorm")
+        else:
+            norm = LayerNorm(elementwise_affine=False, bias=False)
```
Comment (aurelion-source) [inline megatron/model/norms.py:41]: This leads to an error due to initializing torch.nn.LayerNorm without normalized_shape.
Also, get_norm should not return an initialized object.
#1338 resolves this by creating a separate NonParametricLayernorm class.
